torch/_inductor/codegen/cpp.py

```diff
@@ -1468,13 +1472,51 @@ def __init__(self, args, num_threads):
         self.reduction_depth = None
         self.reduction_prefix = IndentedBuffer()
         self.reduction_suffix = IndentedBuffer()
-        self.reduction_var_map = {}
+        self.parallel_reduction_prefix = IndentedBuffer()
+        self.parallel_reduction_suffix = IndentedBuffer()
+        self.parallel_reduction_stores = IndentedBuffer()
+        self.worksharing_reduction_init = IndentedBuffer()
+        self.worksharing_reduction_stores = IndentedBuffer()
```

Comment:
Would "local" be a more proper name here?
```suggestion
        self.local_reduction_init = IndentedBuffer()
        self.local_reduction_stores = IndentedBuffer()
```